src/cli/commands/decrypt.test.ts

```diff
@@ -0,0 +1,150 @@
+      ])
+      .exit(0)
+      .it('shows help when other args are passed alongside --help', ctx => {
+        expect(ctx.stdout.split('\n')).toEqual(expectedHelpOutput)
+      })
+  })
+
+  describe('decrypts successfully', () => {
+    const encryptedConfigPath = 'example/development.yaml'
+    const backupEncryptedConfigPath = 'example/development.backup.yaml'
+    const outputFilePath = 'tmp.decrypted.yaml'
+    const schemaPath = 'example/schema.json'
+
+    beforeAll(() => {
+      // Copy encrypted config file so we can restore it later
+      fs.copyFileSync(encryptedConfigPath, backupEncryptedConfigPath)
+    })
+
+    afterEach(() => {
+      // Delete temporary output file (decrypted config)
+      try {
+        fs.unlinkSync(outputFilePath)
+      } catch (err) {
+        console.log('No temporary output file found to delete')
+      }
+
+      // Delete config file, which might have been decrypted in-place, and restore backup config.
+      fs.unlinkSync(encryptedConfigPath)
+      fs.copyFileSync(backupEncryptedConfigPath, encryptedConfigPath)
+    })
+
+    afterAll(() => {
+      // Delete backup config
+      try {
+        fs.unlinkSync(backupEncryptedConfigPath)
+      } catch (err) {
+        console.log('Could not find backup config file to delete')
+      }
+    })
+
+    describe('decrypts to separate output file', () => {
+      test
+        .command(['decrypt', encryptedConfigPath, outputFilePath])
+        .exit(0)
+        .it(
+          `decrypts successfully and writes to file ${outputFilePath}`,
+          () => {
+            expect(fs.existsSync(outputFilePath)).toBe(true)
+            expect(
+              fs
+                .readFileSync(outputFilePath, { encoding: 'utf8' })
+                .replace(/\s/g, '')
+            ).toBe(expectedDecryptedConfig)
+          }
+        )
+    })
+
+    describe('decrypts in-place when not output file is passed', () => {
+      test
+        .command(['decrypt', encryptedConfigPath])
+        .exit(0)
+        .it('decrypts successfully in-place', () => {
+          expect(
+            fs
+              .readFileSync(encryptedConfigPath, { encoding: 'utf8' })
+              .replace(/\s/g, '')
+          ).toBe(expectedDecryptedConfig)
+        })
+    })
+
+    // TODO: Only tests against terminal output. Better approach would be to mock and
+    // check that the mock (here: the validation function) has been called correctly.
+    describe('decrypts and validates successfully', () => {
+      test
+        .stderr()
+        .command(['decrypt', encryptedConfigPath, '--schema-path', schemaPath])
+        .exit(0)
+        .it('decrypts and validates successfully', ctx => {
+          expect(ctx.stderr).toContain('Decrypting... done')
+          expect(ctx.stderr).toContain('Validating... done')
+        })
+    })
+
+    describe('fails when...', () => {
+      test
+        .command(['decrypt'])
+        .exit(2)
+        .it('no arguments are passed')
```

Comment:
maybe i misunderstand the oclif testing syntax, but this looks like it's not really testing anything to me

Reply:
guessing this is just not implemented yet because PR is still in "draft", if that's the case feel free to resolve this thread @mohoff

Reply:
Generally, this is the way `@oclif/test` is used (https://oclif.io/docs/testing). In this case I wanted to test for exit code 2. IIRC this works, replacing it with `exit(0)` fails the test.

More importantly though there is more stuff broken with the tests. They work on a per-file basis but running the test suits fails. I think I found an alternative testing way though. more soon

Reply:
got it, thx for the link